src/Microsoft.AspNetCore.Razor.Language/Intermediate/DesignTimeDirectiveIRNode.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Razor.Language.CodeGeneration;
+
+namespace Microsoft.AspNetCore.Razor.Language.Intermediate
+{
+    internal class DesignTimeDirectiveIRNode : ExtensionIRNode
+    {
+        public override IList<RazorIRNode> Children { get; } = new List<RazorIRNode>();
+
+        public override RazorIRNode Parent { get; set; }
+
+        public override SourceSpan? Source { get; set; }
+
+        public override void Accept(RazorIRNodeVisitor visitor)
+        {
+            if (visitor == null)
+            {
+                throw new ArgumentNullException(nameof(visitor));
+            }
+
+            AcceptExtensionNode<DesignTimeDirectiveIRNode>(this, visitor);
+        }
+
+        public override void WriteNode(RuntimeTarget target, CSharpRenderingContext context)
+        {
+            var extension = target.GetExtension<IDesignTimeDirectiveTargetExtension>();
```

Comment:
@rynowak Letting this null ref doesn't seem like a great pattern for all of our extensions. How do you feel about exploding with a better error message about a missing extension? If you like, should be a separate issue; not done in this PR.